packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -21,23 +21,69 @@ export default function({ types: t }) {
     return false;
   }
 
-  function createObjectSpread(file, props, objRef) {
-    const restElement = props.pop();
-
+  // normalizes identifier keys to string literals and extracts all keys
+  // e.g. extracts ["a", "b", 3, ++x] from {a: "foo", b, 3: "bar", [++x]: "baz"}
+  function extractNormalizedKeys(path) {
+    const propPaths = path.get("properties");
     const keys = [];
-    for (const prop of props) {
-      let key = prop.key;
-      if (t.isIdentifier(key) && !prop.computed) {
-        key = t.stringLiteral(prop.key.name);
+    for (const propPath of propPaths) {
+      const key = propPath.get("key").node;
+      if (t.isIdentifier(key) && !propPath.node.computed) {
+        keys.push(t.stringLiteral(key.name));
+      } else {
```

Comment:
I mean specifically for

```js
{ 0: a, ...b }
```

In this case, the `0` NumericLiteral can be transformed into a string already. But the location you had it (inside an `if (isIdentifier)`) won't work, because literals aren't identifiers.